### DISCUSSION TITLE
Add an 'Apply' button to the 'Datetime' field Date/Time selector

### DISCUSSION THREAD

**Discussion by wfriedman89** [directus/directus#24899] [Feature Requests]
### Describe the Improvement

It would be nice to add an apply button to the 'Datetime' field Date/Time selector.  It could be next to the 'Set to Now' button

The prominence of the 'Set to Now' button could be confused with an 'Apply' as the only way to get that selector to go away right now is to click off off it.

![Screenshot 2024-03-28 at 11 42 08 AM](https://github.com/directus/directus/assets/109602228/4845ae36-000d-4e9d-baa3-44d228e3c11d)

**Comment by shipsides-sparknel** [comment]
to piggy back on any potential changes to the picker, could a change to the month and year selector be considered?

the current method is "fine" for changing the month within the current year, but to change the year requires the selection and edit of the text, which isn't an easy user experience especially on mobile.

https://demos.telerik.com/kendo-ui/datepicker/index is a great example of ideal UX - selecting April 2024 then shows you all the months in 2024, selecting 2024 then shows you +/- 5 years from 2024, then selecting 2019-2029 shows you +/- 5 decades. it makes adding things like dates of birth simple and intuitive.

**Comment by hanneskuettner** [comment]
A few considerations before we implement this:

- When adding the "Apply" button, we would also need a "Cancel" button, otherwise the wording on "Apply" is misleading as I would expect changes to the date selection to only be applied on clicking the button, but discarded otherwise.
- As a direct result clicking away from the data picker should act the same as the "Cancel" button, changing the current UX, confusing users used to being able to click away and still change the date.

Two alternatives could be:
- Add a `Close" button, that just closes the date picker, offering another level of confusing UX to the user.
- Making the "Set to now" button less prominent by inlining it with the time selection, sticking with the current UX.

I would appreciate input on this.
